modules/configuring-default-route-timeouts-globally.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+//
+// * knative-serving/config-applications/configuring-revision-timeouts.adoc
+:_content-type: PROCEDURE
+[id="configuring-default-route-timeouts-globally_{context}"]
+= Configuring the default route timeouts globally
+
+By configuring the route timeouts globally, you can ensure consistent timeout settings across all services, simplifying management for workloads that have similar timeout needs and reducing the need for individual adjustments.
+
+You can configure the route timeouts globally by updating the `ROUTE_HAPROXY_TIMEOUT` environment value in your `serverless-operator` subscription and updating the `max-revision-timeout-seconds` in your `KnativeServing` custom resource (CR). This applies the timeout changes across all Knative services, and you can deploy services with specific timeouts up to the maximum value set.
+
+.Procedure
+
+. Set the value of `ROUTE_HAPROXY_TIMEOUT` in your subscription to your required timeout in seconds:
++
+.Example of setting the `ROUTE_HAPROXY_TIMEOUT` value to 900 seconds
+[source,terminal]
+----
+$ oc patch subscription.operators.coreos.com serverless-operator -n openshift-serverless --type='merge' -p '{"spec": {"config": {"env": [{"name": "ROUTE_HAPROXY_TIMEOUT", "value": "900"}]}}}'
+----
++
+`ROUTE_HAPROXY_TIMEOUT` is an enviroment managed by the Serverless Operator and by default is set to `600`. This subscription update causes pods to be redeployed in the `openshift-serverless` namespace.
+
+[NOTE]
+====
+If you created your routes manually and disabled auto-generation with the annotation `serving.knative.openshift.io/disableRoute`, you can configure the timeouts directly in the route definitions.
+====
+
+. Set the maximum revision timeout as follows in the `KnativeServing` custom resource (CR):
++
+.Example of `max-revision-timeout-seconds` set to 900 seconds
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+spec:
+  config:
+    defaults:
+      max-revision-timeout-seconds: "900"
+----
++
+The Serverless Operator automatically adjusts the activator's `terminationGracePeriod` to the set maximum revision timeout value to avoid request termination in cases where activator pods are being terminated themselves.
```

Comment:
```suggestion
The Serverless Operator automatically adjusts the `terminationGracePeriod` value of the activator to the set maximum revision timeout value to avoid request termination in cases where activator pods are being terminated themselves.
```

https://www.ibm.com/docs/en/ibm-style?topic=grammar-possessives
> Do not use possessive ’s with inanimate objects.